mongo/src/test/java/com/redhat/lightblue/mongo/crud/UpdateIfSameTest.java

```diff
@@ -0,0 +1,227 @@
+        BasicDBObject doc=new BasicDBObject("_id",id).append("field",value);
+        if(versions!=null&&versions.length>0) {
+            doc.append(DocTranslator.HIDDEN_SUB_PATH.toString(),
+                       new BasicDBObject(DocVerUtil.DOCVER,new ArrayList<ObjectId>(Arrays.asList(versions))));
+        }
+        return doc;
+    }
+    
+    @Test
+    public void successfulUpdateTest() throws Exception {
+
+        DBObject doc;
+
+        doc=getDoc("1","1",new ObjectId());
+        DocIdVersion ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+
+        doc=getDoc("2","2",new ObjectId());
+        ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+        
+        doc=getDoc("3","3",new ObjectId());
+        ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+        
+        doc=getDoc("4","4",new ObjectId());
+        ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+        
+        Assert.assertTrue(updater.commit().isEmpty());
+
+        DBCursor cursor=coll.find();
+        while(cursor.hasNext()) {
+            doc=cursor.next();
+            Assert.assertEquals("updated"+doc.get("_id").toString(),doc.get("field").toString());
+            Assert.assertEquals(updater.docVer.toString(),((List<ObjectId>)((DBObject)doc.get(DocTranslator.HIDDEN_SUB_PATH.toString())).
+                                                           get(DocVerUtil.DOCVER)).get(0).toString());
+        }
+        cursor.close();
+        
+    }
+
+    @Test
+    public void dontUpdateIfVersionUnknown() throws Exception {
+
+        DBObject doc;
+
+        doc=getDoc("1","1",new ObjectId());
+        DocIdVersion ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+
+        doc=getDoc("2","2",new ObjectId());
+        ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+        
+        doc=getDoc("3","3",new ObjectId());
+        ver=DocIdVersion.getDocumentVersion(doc);
+        coll.insert(doc);
+        updater.addVersion(ver);
+        updater.addDoc(doc);
+        doc.put("field","updated"+doc.get("_id").toString());
+
+        // Id:4 will not be updated, its version is unknown
```

Comment:
I don't understand the question. Version is always known, it is either null, or the value at the top of the hidden array in the doc.